streaming/base/storage/download.py

```diff
@@ -355,17 +360,34 @@ def download_from_dbfs(remote: str, local: str) -> None:
     """
     try:
         from databricks.sdk import WorkspaceClient
+        from databricks.sdk.core import DatabricksError
     except ImportError as e:
         e.msg = get_import_exception_message(e.name)  # pyright: ignore
         raise e
 
+    obj = urllib.parse.urlparse(remote)
+    if obj.scheme != 'dbfs':
+        raise ValueError(f'Expected path prefix to be `dbfs` if it is a Databricks File System, ' +
+                         f'instead, got {obj.scheme} for remote={remote}.')
+
     client = WorkspaceClient()
-    file_path = remote.lstrip('dbfs:')
-    dbfs_file = client.dbfs.download(file_path)
+    file_path = urllib.parse.urlparse(remote)
     local_tmp = local + '.tmp'
-    with open(local_tmp, 'wb') as f:
-        f.write(dbfs_file.read())
-    dbfs_file.close()
+    try:
+        with client.dbfs.download(file_path.path) as response:
+            with open(local_tmp, 'wb') as f:
+                # Multiple shard files are getting downloaded in parallel, so we need to
+                # read the data in chunks to avoid memory issues.
+                # Read 1MB of data at a time since that's the max limit it can read at a time.
```

Comment:
this is pretty small, curious if this bottlenecks training down the line?

Reply:
Not in this PR, but we should have a benchmark for this.

Reply:
This is the max limit they have. Maybe this has a minor impact on the start up time but not much during model training as we download shard files ahead of time, plus, the recommendation is to use UC Volume rather than DBFS. Hence, I don't see much usage of DBFS in general.